app/controllers/scim_rails/application_controller.rb

```diff
@@ -35,5 +35,61 @@ def authenticate_with_oauth_bearer
 
       yield searchable_attribute, authentication_attribute
     end
+    
+    # Shared stuff...
+    def find_value_for(attribute)
+      params.dig(*path_for(attribute))
+    end
+
+    # `path_for` is a recursive method used to find the "path" for
+    # `.dig` to take when looking for a given attribute in the
+    # params.
+    #
+    # Example: `path_for(:name)` should return an array that looks
+    # like [:names, 0, :givenName]. `.dig` can then use that path
+    # against the params to translate the :name attribute to "John".
+    def path_for(attribute, object = ScimRails.config.mutable_user_attributes_schema, path = [])
+      at_path = path.empty? ? object : object.dig(*path)
+      return path if at_path == attribute
+
+      case at_path
+      when Hash
+        at_path.each do |key, value|
+          found_path = path_for(attribute, object, [*path, key])
+          return found_path if found_path
+        end
+        nil
+      when Array
+        at_path.each_with_index do |value, index|
+          found_path = path_for(attribute, object, [*path, index])
+          return found_path if found_path
+        end
+        nil
```

Comment:
🌵 What are these `nil` fallbacks doing for you?

Reply:
This was also something I only moved around